File: swift/Sources/SignalClient/zkgroup/AuthCredentialResponse.swift
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation
import SignalFfi

public class AuthCredentialResponse: ByteArray {

  public static let SIZE: Int = 361

  public required init(contents: [UInt8]) throws {
    try super.init(newContents: contents, expectedLength: AuthCredentialResponse.SIZE)

    try withUnsafePointerToSerialized { contents in
      try checkError(signal_auth_credential_response_check_valid_contents(contents))
    }
  }

}

File: java/java/src/main/java/org/signal/zkgroup/auth/AuthCredentialResponse.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.auth;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class AuthCredentialResponse extends ByteArray {

  public static final int SIZE = 361;

  public AuthCredentialResponse(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
    Native.AuthCredentialResponse_CheckValidContents(contents);
  }

  public byte[] serialize() {
    return contents.clone();
  }

}
